File: fundamentos/desafio-js/src/questions/get-client-w-lowest-balance-at-ag-47.ts
import { IAccount } from "../i-account";

/** O nome do(a) cliente com o menor saldo na agência 47 é: */
export default function getClientWLowestBalanceAtAg47(...accounts: Array<IAccount>) {
  const acc = accounts
    .filter(acc => acc.agencia == 47)
    .sort((a, b) => a.balance - b.balance)[0];

  console.info(`O nome do(a) cliente com o menor saldo na agência 47 é: ${acc.name}, $${acc.balance}`);
}

File: fundamentos/desafio-dart/lib/questions/get_client_w_lowest_balance_at_ag_47.dart
import 'package:desafio_dart/domains/account.dart';
import 'package:desafio_dart/questions/get_client_w_highest_balance_at_ag_10.dart';

/// O nome do(a) cliente com o menor saldo na agência 47 é:
void getClientWLowestBalanceAtAg47(List<Account> accounts) {
  final accs = accounts.where((item) => item.agencia == 47).toList();

  accs.sort((a, b) => a.balance.compareTo(b.balance));

  final acc = accs.first;

  print(
      'O nome do(a) cliente com o menor saldo na agência 47 é: ${acc.name}, \$${acc.balance}');
}
